split-llvm-extract.cpp

```diff
@@ -271,15 +271,27 @@ int main(int argc, char **argv)
 #pragma omp single
 	for (llvm::GlobalVariable &globalVariable : loadedModule->globals())
 	{
-		if (globalVariable.isConstant() || globalVariable.isExternallyInitialized())
+		std::string globVar = globalVariable.getName().str();
+
+		if (globalVariable.isExternallyInitialized())
+		{
+			continue;
+		}
+
+		if (globalVariable.isConstant() && globalVariable.hasInitializer() &&
```

Comment:
I think we can simplify lines 274-290 to something like:

```c
string glob_name;
if (globalvariable.isconstant() && globalvariable.hasinitializer()) {
  if (!globalvariable.getinitializer()->hashname()) {
    continue;
  }
  glob_name = globalVariable.getInitializer()->getName().str();
} else {
  glob_name = globalVariable.getName().str();
}
```

I think it would be better not to have variables called `globalVariable` and `globVar` -- I read them as synonyms at first! `globName` or similar might work instead?

Reply:
Done in 27dab17.